WcaOnRails/app/javascript/edit-schedule/EditSchedule.jsx

```diff
@@ -0,0 +1,218 @@
+                  />
+                </Panel.Body>
+              </Panel>
+              <Panel id="schedules-edit-panel" bsStyle="info" eventKey="2">
+                <div id="accordion-schedule-heading-2" className="panel-heading heading-as-link" aria-controls="accordion-schedule-body-2" role="button" data-toggle="collapse" data-target="#accordion-schedule-body-2" data-parent="#accordion-schedule">
+                  <Panel.Title>
+                    Edit schedules <span className="collapse-indicator"></span>
+                  </Panel.Title>
+                </div>
+                <Panel.Body id="schedules-edit-panel-body" collapsible>
+                  <SchedulesEditor scheduleWcif={scheduleWcif} eventsWcif={competitionInfo.eventsWcif} locale={locale} />
+                </Panel.Body>
+              </Panel>
+            </PanelGroup>
+          </Col>
+        </Row>
+        {unsavedChanges}
+      </div>
+    );
+  }
+}
+
+const UnsavedChangesAlert = ({ actionHandler, saving }) => {
+  return (
+    <Alert bsStyle="info">
+      You have unsaved changes. Don't forget to{" "}
+      <button onClick={actionHandler}
+        disabled={saving}
+        className={cn("btn", "btn-default btn-primary", { saving: saving })}
+      >
+        save your changes!
+      </button>
+    </Alert>
+  );
+}
+
+const IntroductionMessage = () => {
+  return (
+    <Col xs={12}>
+      <p>
+        Depending on the size and setup of the competition, it may take place in several rooms of several venues.
+        Therefore a schedule is necessarily linked to a specific room.
+        Each room may have its own schedule (with all or a subset of events).
+        You can create the competition's schedule below by first creating at least one venue with one room first.
+        Then you will be able to select this room in the "Edit schedules" panel, and drag and drop event rounds (or attempts for some events) on it.
+      </p>
+      <p>
+        For the typical simple competition, creating one "Main venue" with one "Main room" is enough.
+        If your competition has a single venue but multiple "stages" with different schedules, please input them as different rooms.
+      </p>
+    </Col>
+  );
+}
+
+const VenuesList = ({venues, actionsHandlers, competitionInfo}) => {
+  return (
+    <Row>
+      {venues.map((venueWcif, index) => {
+        return (
+          <EditVenue
+            venueWcif={venueWcif}
+            key={index}
+            index={index}
+            removeVenueAction={e => actionsHandlers.removeVenue(e, index)}
+            competitionInfo={competitionInfo}
+          />
+        );
+      })}
+      <NewVenue actionHandler={actionsHandlers.addVenue} />
+    </Row>
+  );
+}
+
+const NewVenue = ({ actionHandler }) => {
+  return (
+    <div className="panel-venue">
+      <a href="#" className="btn btn-success new-venue-link" onClick={actionHandler}>Add a venue</a>
+    </div>
+  );
+}
```

Comment:
Functional components can be written nicely without explicit return (same goes for all the others), just like so:
```jsx
const NewVenue = ({ actionHandler }) => (
  <div className="panel-venue">
    <a href="#" className="btn btn-success new-venue-link" onClick={actionHandler}>Add a venue</a>
  </div>
);
```
Shorter and less indentation :tada: